module/applications/dice/roll-configuration-dialog.mjs

```diff
@@ -121,6 +130,48 @@ export default class RollConfigurationDialog extends Application5e {
   /*  Rendering                                   */
   /* -------------------------------------------- */
 
+  /**
+   * Identify DiceTerms in this app's rolls.
+   * @returns {{ icon: string, label: string }[]}
+   * @protected
+   */
+  _identifyDiceTerms() {
+    const dice = [];
+    let shouldDisplay = true;
+
+    /**
+     * Determine if a given term is displayable.
+     * @param {RollTerm} term  The term.
+     * @returns {boolean|void}
+     */
+    const identifyTerm = term => {
+      if ( !(term instanceof DiceTerm) ) return;
+      // If any of the terms have complex components, do not attempt to display only some dice, bail out entirely.
+      if ( !Number.isFinite(term.number) || !Number.isFinite(term.faces) ) return shouldDisplay = false;
+      for ( let i = 0; i < term.number; i++ ) dice.push({
+        icon: `systems/dnd5e/icons/svg/dice/${term.denomination}.svg`,
```

Comment:
Gonna look real odd when rolling d2s and d7s. Should set `shouldDisplay = false` if they have an invalid denomination too.